pandas/core/resample.py

```diff
@@ -2457,8 +2457,15 @@ def _get_timestamp_range_edges(
     """
     if isinstance(freq, Tick):
         index_tz = first.tz
-        if isinstance(origin, Timestamp) and (origin.tz is None) != (index_tz is None):
-            raise ValueError("The origin must have the same timezone as the index.")
+        if isinstance(origin, Timestamp) and origin.tz != index_tz:
+            if first.utcoffset() != origin.utcoffset():
+                raise ValueError("The origin must have the same timezone as the index.")
```

Comment:
would this break during daylight savings changes?
```python
In [39]: Timestamp('2020-10-25', tz='Europe/London').utcoffset()
Out[39]: datetime.timedelta(seconds=3600)

In [40]: Timestamp('2020-10-26', tz='Europe/London').utcoffset()
Out[40]: datetime.timedelta(0)
```

Reply:
good catch